### PR TITLE
Fix OpenSSL static dependencies.

### LINKED ISSUE #458: Libssh2 Linux build failed
@alamaison I've push latest version to testing and see some failed stuff: https://travis-ci.org/ingenue/hunter/builds/145236987 I don't know what changed (it was working before), probably this is because of FindOpenSSL update (?). I think we need to link CMAKE_DL_LIBS to OpenSSL::crypto. But first we need to use imported OpenSSL targets in Libssh2 instead of OPENSSL_LIBRARIES as far as I understand: https://github.com/hunter-packages/libssh2/blob/266056ce760d23998e60a4bb6064362257973215/src/CMakeLists.txt#L70

Needed for CURL: https://github.com/ruslo/hunter/pull/456

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/36084753-libssh2-linux-build-failed?utm_campaign=plugin&utm_content=tracker%2F1032790&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1032790&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alamaison**
Fixes #458.

### CODE REVIEW

File: cmake/find/FindOpenSSL.cmake [hunk 1]
```diff
@@ -254,7 +254,8 @@ else()
   set(OPENSSL_SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY})
   set(OPENSSL_CRYPTO_LIBRARIES ${OPENSSL_CRYPTO_LIBRARY})
 
-  set(OPENSSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
+  set(OPENSSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY} ${CMAKE_DL_LIBS})
+)
```
Comment (ruslo) [inline cmake/find/FindOpenSSL.cmake]: Syntax error

Comment (alamaison) [inline cmake/find/FindOpenSSL.cmake]: Whoops.  Not sure what happened there.  Fixed now

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.792
* https://travis-ci.org/ingenue/hunter/builds/152229258

**Comment by alamaison** [comment]
The same fix was needed for the OpenSSL::Crypto and OpenSSL::SSL targets.  I've updated the PR

**Comment by ruslo** [comment]
> The same fix was needed for the OpenSSL::Crypto and OpenSSL::SSL targets

This change affects cache. To show the fact that we need to use another cache entry even if no cmake arguments/url/sha1 changed need to add `PACKAGE_INTERNAL_DEPS_ID 1` to OpenSSL/hunter.cmake. See example: https://github.com/ruslo/hunter/blob/94337ffdd4aee76bbbc37a5085031dc63d9464e2/cmake/projects/GTest/hunter.cmake#L156

**Comment by alamaison** [comment]
Updated

**Comment by ruslo** [comment]
OpenSSL testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.794
* https://travis-ci.org/ingenue/hunter/builds/152234008

Libssh2 testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.793
* https://travis-ci.org/ingenue/hunter/builds/152232963

**Comment by alamaison** [comment]
MSYS build failed with "The source directory "C:/projects/hunter/examples/Foo" does not exist."

**Comment by ruslo** [comment]
> MSYS build failed with

There is no `Foo` test. I fixed it, particularity of config updates while doing `git merge`.

Uploading OpenSSL (libssh2 is not cacheable, no upload):
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.795
* https://travis-ci.org/ingenue/hunter/builds/152247974

**Comment by ruslo** [comment]
> Uploading OpenSSL

Done. Released: https://github.com/ruslo/hunter/releases/tag/v0.15.15

**Pull request merged**